packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -41,9 +43,19 @@ export interface ViewCustomTimings {
 export const THROTTLE_VIEW_UPDATE_PERIOD = 3000
 export const SESSION_KEEP_ALIVE_INTERVAL = 5 * ONE_MINUTE
 
-export function trackViews(location: Location, lifeCycle: LifeCycle) {
+export function trackViews(location: Location, lifeCycle: LifeCycle, newLocationListener?: NewLocationListener) {
   const startOrigin = 0
-  const initialView = newView(lifeCycle, location, ViewLoadingType.INITIAL_LOAD, document.referrer, startOrigin)
+  const onNewLocation: NewLocationListener = newLocationListener || (() => undefined)
+
+  const { viewName } = onNewLocation(location) || {}
```

Comment:
Since `onNewLocation` is defined by customers, how should we behave if it throw?

Reply:
I think we should catch the error, create the view then add the error to it as custom error. WDYT?

Reply:
For other cases where we want to communicate a wrong configuration to the customer, we use a console.warn (ex: add timing). 
I'd be in favor of still using the console (easier to see during the integration), but promoting this one to error, it will still be collected.